src/backend/distributed/planner/multi_planner.c

```diff
@@ -621,23 +708,65 @@ multi_relation_restriction_hook(PlannerInfo *root, RelOptInfo *relOptInfo, Index
 }
 
 
+/* GetRTEIdentity returns the identity assigned with AssignRTEIdentity. */
+int
+GetRTEIdentity(RangeTblEntry *rte)
+{
+	Assert(rte->rtekind == RTE_RELATION);
+	Assert(IsA(rte->values_lists, IntList));
+	Assert(list_length(rte->values_lists) == 1);
+
+	return linitial_int(rte->values_lists);
+}
+
+
 /*
- * CreateAndPushRestrictionContext creates a new restriction context, inserts it to the
- * beginning of the context list, and returns the newly created context.
+ * CopyPlanParamList deep copies the input PlannerParamItem list and returns the newly
+ * allocated list.
+ * Note that we cannot use copyObject() function directly since there is no support for
+ * copying PlannerParamItem structs.
  */
-static RelationRestrictionContext *
-CreateAndPushRestrictionContext(void)
+static List *
+CopyPlanParamList(List *originalPlanParamList)
```

Comment:
I realize we need to replace params, but why do we need to do so in-place, rather than just in a clause passed to pruning?